Bindings/Java/Matlab/Utilities/osimTableFromStruct.m

```diff
@@ -20,108 +26,91 @@
-        nColsList = [nColsList nCol];
-    end
-    
-    % check to see if nRows is uniform
-    if ~isempty(find( nRowsList ~= nRowsList(1)) )
-        error(['Number of rows is non-uniform across fields. Check .' labelnames{i} ' for error'])
     end
-    
-    % check to see if nRows is uniform
-    if ~isempty(find( nColsList ~= nColsList(1)) )
-        error(['Number of columns is non-uniform across fields. Check .' labelnames{i} ' for error'])
-    end        
 end
-  
-%% make an array of index numbers that will be used in the main For loop
-indexArray = [0 : length(labelnames) - 1];
-indexArray(timeIndex) = [];
-nLabels = length(indexArray);
 
-%% Pre-allocate an empty data table from data type. Also set some flags. 
-if max(nColsList) == 3
-    vec3table = true;
-    osimtable = DataTableVec3();
-else
-    vec3table = false;
+%% Pre-allocate an empty data table from data type. 
+if colRef == 1
     osimtable = DataTable();
+else
+    osimtable = DataTableVec3();
 end
 
 %% set the osimtable column names
-labels =  StdVectorString();
-for i = indexArray
-    labels.add( labelnames{i+1} );
+osimlabels =  StdVectorString();
+for i = 1 : nfields
+    osimlabels.add( labels{i} );
 end
-osimtable.setColumnLabels(labels);
-
-%% get a pointer to the time column
-timeColumn = osimtable.getIndependentColumn();
+osimtable.setColumnLabels(osimlabels);
 
-if vec3table == true
-    % If the data is nX3, then build an osimtable of Vec3's.
-    % We build the osimtable row by row, filling a row vector from left to
-    % right and then appending the row to the OpenSim table.
-   for iRow = 1 : nRows
-        % elems will be a vec3 vector that we will append elements to
-        elems = StdVectorVec3();
+%% Build the osimtable row by row, filling a row vector from left to
+% right and then appending the row to the OpenSim osimtable. 
+if colRef == 1
+    % Get an OpenSim Row Vector
+    row = RowVector(nfields, NaN);
+    % create and fill a row of data 
+    for iRow = 1 : nRows 
+        for iCol = 1 : nfields 
+           % Get the row value of the array from each field and set that
+           % value in the OpenSim Row
+           row.set(iCol-1, s.(labels{iCol})(iRow) );
+        end
+        % Append the row vector to the opensim table
+        osimtable.appendRow(iRow-1, row);
+    end
+else
+    % Get an empty Vec3 
+	elems = StdVectorVec3();
+    % Fill rows and elements with 
+    for iRow = 1 : nRows
         % create and fill a row of data
-        for iCol = indexArray
+        for iCol = 1 : nfields 
             % get the data from the input structure
-            rowdata = structdata.(labelnames{iCol+1})(iRow,:);
+            elemtdata = s.(labels{iCol})(iRow,:);
             % make a vec3 element from the rowdata
-            elem = Vec3(rowdata(1), rowdata(2), rowdata(3));
+            elem = Vec3(elemtdata(1), elemtdata(2), elemtdata(3));
             % append the Vec3 element to the vec3 vector
             elems.add(elem);
```

Comment:
As-is, I think `elem` will keep growing with each row. Perhaps copy what you did earlier: set the size of `StdVectorVec3` then index into `elems`, rather than using `add()`.

Reply:
This is a great suggestion, thanks.